src/types/plugin-input.ts

```diff
@@ -41,6 +41,7 @@ export const pluginSettingsSchema = T.Object(
     taskStaleTimeoutDuration: T.String({ default: "30 Days" }),
     startRequiresWallet: T.Boolean({ default: true }),
     maxConcurrentTasks: maxConcurrentTasks(),
+    checkAssignedIssues: T.String({ default: "org" }),
```

Comment:
Usually it's better to use 

```ts
const AssignedIssueScope = {
  ORG: "org",
  REPO: "repo",
  NETWORK: "network"
} as const
```

over enums, they are debated to be a bad practice and TS themselves said they regret adding it into the language. It results in less performant JS (because enums don't exist in JS) and can be lost during runtime in libs are not built correctly

They both represent a simple object where `Enum` adds a burden that can be avoided but it's dev preference in this regard as it's a point of debate in the TS world.

Reply:
TS enum is better in the way it also creates a type, but in your case you have to do `typeof AssignedIssueScope[keyof typeof AssignedIssueScope]`. I think `const enum` basically generates an object like yours

Reply:
https://www.typescriptlang.org/docs/handbook/enums.html#const-enum-pitfalls

const enum is fine here I'm sure because this plugin is not used as a lib but if it was and placed inside a `.ts` file it would be lost at compile time after the file was converted into `.d.ts`. The values will be inlined where they are used.

> Const enums can only use constant enum expressions and unlike regular enums they are completely removed during compilation. Const enum members are inlined at use sites.

> TS enum is better in the way it also creates a type, but in your case you have to do typeof AssignedIssueScope[keyof typeof AssignedIssueScope]

I agree it's a cleaner type definition it just has gotchas with using them so I like to avoid, but dev preference for sure I'm not against them.

Reply:
[TS Playground](https://www.typescriptlang.org/play/#code/MYewdgzgLgBAghCBLA5mApgEwJKIK7oDKoADujALwwDeAUDDAPIBKA4gFwwBEIATilwA09GMwCiABUacuvdCRBCRAOTEAVAOosA0jIxQA7nwDWXWgF8YAQwgxQkKLXvQY0eQEZK3NyXdmn4C7oYHgAtvCIqBg4+ESk6ADCgVBiIeF0DCysXjz8SgziUjlyCvkwqpo6OfpGvKYWAQ6uUPIATDk+rf60wWERyGhYuBAExCBkqX0ZTGw5fALCBZKMxfKKi+XqWsza1eiGJmbmtLRAA)

this shows how the three enums are compiled